src/analog/m2kanalogin_impl.cpp

```diff
@@ -105,55 +91,60 @@ void M2kAnalogInImpl::init()
 
 	for (unsigned int i = 0; i < getNbChannels(); i++) {
 		enableChannel(i, true);
+		auto ch = static_cast<ANALOG_IN_CHANNEL>(i);
 
-		ANALOG_IN_CHANNEL ch = static_cast<ANALOG_IN_CHANNEL>(i);
 		m_trigger->setAnalogMode(ch, ALWAYS);
-
 		setRange(ch, PLUS_MINUS_25V);
-		m_adc_calib_offset.at(i) = 2048;
-		m_adc_calib_gain.at(i) = 1;
-		m_adc_hw_vert_offset.at(i) = 0;
-		m_adc_hw_vert_offset_raw.at(i) = 2048;
+		setAdcCalibOffset(ch, 2048);
+		setAdcCalibGain(ch, 1);
+		setVerticalOffset(ch, 0);
 		m_trigger->setCalibParameters(i, getScalingFactor(i), m_adc_hw_vert_offset.at(i));
 	}
-	setKernelBuffersCount(1);
+	setKernelBuffersCount(4);
```

Comment:
I think this value `4` could be stored in a member variable, or static, maybe called `default_kernel_buffers` or w/e the naming convention is for the project.